### ISSUE TITLE
Catching SystemExit in kernel loop

### ISSUE THREAD

**Issue by almarklein** [pyzo/pyzo#316]
Originally reported by: **rreilink (Bitbucket: [rreilink](http://bitbucket.org/rreilink), GitHub: [rreilink](http://github.com/rreilink))**

---

When SystemExit is raised, the kernel should not quit but instead stop the program and show the prompt >>>

This could be done by catching SystemExit and then printing a message. In issue #250 it was discussed wheteher there could be an 'intentional' exit() that should stop the kernel. I think this is not necessary since it would mimick the behaviour of the plain Python interpreter in interactive mode (-i command line switch)

E.g. if we have test.py:

```
print(1)
import sys
sys.exit()
print(2)
```

And then run it from the command prompt in interactive mode:

```
C:\tmp>python -i test.py
1
Traceback (most recent call last):
  File "test.py", line 3, in <module>
    sys.exit()
SystemExit
>>> 
```

---
- Bitbucket: https://bitbucket.org/iep-project/iep/issue/316

**Comment by rreilink** [comment]
Specific use case would be debugging / developing of any code that uses a sys.exit(mainfunction()) construct.

Would it be a problem to always catch SystemExit? Currently, the user ends with a dead shell; if we always catch SystemExit the user ends with a terminated code and a SystemExit message

**Comment by almarklein** [comment]
Maybe not. But if we do it like this, we should print a message that saying something like SystemExit has been detected but we do not exit because this is an interactive session.

**Comment by rreilink** [comment]
Agree, a descriptive message is required. Shall i go ahead and create a pr?
Op 10 okt. 2016 11:20 schreef "Almar Klein" <notifications@github.com>:

**Comment by almarklein** [comment]
Yes, please! Nice to see you coding here again :)
